### PR TITLE
[WIP] fix - TorchModule - __setattr__

### PR DESCRIPTION
**Pull request by RishabhDahale**
## Description
This PR is for issue [#5036](https://github.com/OpenMined/PySyft/issues/5036). Looking for guidance @tudorcebere 

## Affected Dependencies
None.

## How has this been tested?
Have ran [this test](https://github.com/OpenMined/PySyft/blob/55bb023f6cc1f1778dc5f5354ab681c4028c3ed9/tests/syft/lib/torch/module_test.py#L83) to test the changes.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
Hi @RishabhDahale this issue is that the current code works, and your code change breaks the remote model using pointers.

```diff
- if "torch.nn" in full_name_with_qualname(klass=type(value)):
+ if isinstance(value, torch.nn.Module):
```


What we probably need, is to have two checks:
`if is normal torch module` or `if is syft pointer klass`.

The reason why this ticket exists is because checking some strings in the class name is not a very safe way to code.
But its needed for these two examples:

```
assert "torch.nn" in "torch.nn.Conv2"
assert "torch.nn" in "syft.proxy.torch.nn.Conv2Pointer"
```

As you can see both need to work and they both have `torch.nn` in them.

**Comment by tudorcebere** [comment]
Closing this due to inactivity, please open a new PR if you still want to fix the issue.
